### PR TITLE
reconnect jitter configurable

### LINKED ISSUE #445: NATS reconnect jitter incorrect
**Describe the bug**

We configure the NATS connection with a reconnect jitter like so:

https://github.com/knative-extensions/eventing-natss/blob/c5c5a9c622f94aea44e9febfccb9207d0aab5d39/pkg/common/nats/conn.go#L96

This option on the NATS SDK is described as so:

```go
// ReconnectJitter is an Option to set the upper bound of a random delay added ReconnectWait.
// Defaults to 100ms and 1s, respectively.
func ReconnectJitter(jitter, jitterForTLS time.Duration) Option {
	return func(o *Options) error {
		o.ReconnectJitter = jitter
		o.ReconnectJitterTLS = jitterForTLS
		return nil
	}
}
```

I believe our implementation is assuming the first parameter is the number of "units" to jitter by, and the second param is the unit, but it's actually two separate jitters for plaintext and TLS connections. The defaults are 100ms and 1s respectively, but our implementation is set to 1000ms and 1ms.

 **Expected behavior**

These should be configurable by the user as separate configuration items.

@astelmashenko this was added in #408, was there any intentional rational for changing the plaintext jitter from the default 100ms to 1000ms? Happy to keep 1000ms as the default if this isn't configured, but I feel like that's rather large and would like to reduce the default if there was no purpose to the increase.

### PR DESCRIPTION
**Pull request by astelmashenko**
Fixes #445

**Release Note**

```release-note
Add additional connection options: reconnectJitterMilliseconds, reconnectJitterTLSMilliseconds
```

### CODE REVIEW

File: pkg/common/nats/conn.go [hunk 1]
```diff
@@ -82,6 +87,9 @@ func NewNatsConn(ctx context.Context, config commonconfig.EventingNatsConfig) (*
 	// reconnection options
 	if config.ConnOpts != nil && config.ConnOpts.RetryOnFailedConnect {
 		reconnectWait := time.Duration(config.ConnOpts.ReconnectWaitMilliseconds) * time.Millisecond
+		reconnectJitter := defaultJitterIfEmpty(config.ConnOpts.ReconnectJitterMilliseconds, defaultReconnectJitter)
+		reconnectJitterTLS := defaultJitterIfEmpty(config.ConnOpts.ReconnectJitterMilliseconds, defaultReconnectJitterTLS)
```
Comment (dan-j) [inline pkg/common/nats/conn.go:91]: typo on the `ConnOpts` field...

```suggestion
		reconnectJitterTLS := defaultJitterIfEmpty(config.ConnOpts.ReconnectJitterTLSMilliseconds, defaultReconnectJitterTLS)
```

**Comment by creydr** [comment]
Hi @astelmashenko,
is there a reason, why you're targeting directly 1.10 branch and not main and then backporting it?

**Comment by astelmashenko** [comment]
This is our current production version. I will cerry-pick this PR into main.

**Pull request merged**